### DISCUSSION TITLE
How to remove password for opening pdf? I have the password, but want to remove password with python.

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#3003] [Q&A] [answered]
How to remove password for opening pdf? I have the password, but want to remove password with python.

**Comment by JorjMcKie** [comment]
Open the PDF, use `doc.authenticate()` with the password (must be the owner password), then save to a new file.

**Comment by nissansz** [comment]
It works.
Thanks.

Two more questions.
1. Some files has this "Enable Editing" warning, how to enable with pympdf.
![image](https://github.com/pymupdf/PyMuPDF/assets/41010669/a65f3573-6651-4f99-b029-4f392ffe1a86)

2. For some files, when opened, it is "enable scrolling", but for other files, this option is not on, will jump to next page when you scroll. 
How to enable scrolling for alll pdf files as a default option through pymupdf?
![image](https://github.com/pymupdf/PyMuPDF/assets/41010669/fbbcdf91-3fd9-49fb-b397-96de0e2480f1)

**Comment by JorjMcKie** [comment, marked as answer]
This is a mixture of different things: available PDF properties plus features of the viewer you are using (which may or may not correspond to PDF properties).
With PyMuPDF you can only set PDF properties of course. This comprises the following:
1. PDF permissions: an integer consisting of bit flags explained [here](https://pymupdf.readthedocs.io/en/latest/document.html#Document.permissions) and to be used in method [save](https://pymupdf.readthedocs.io/en/latest/document.html#Document.save). Whether your viewer means [PDF_PERM_MODIFY](https://pymupdf.readthedocs.io/en/latest/vars.html#document-permissions) when it says "editing" ... who knows.
2. PDF page mode: what to show when viewing the PDF (TOC Yes/No, embedded files, etc). See [here](https://pymupdf.readthedocs.io/en/latest/document.html#Document.set_pagemode).
3. PDF page layout: things like single or double pages, etc. See [here](https://pymupdf.readthedocs.io/en/latest/document.html#Document.set_pagelayout).

You must try out how your viewer translates these PDF terms.

**Comment by nissansz** [comment]
Thank you.
